### PR TITLE
blst 0.3.11 (new formula) slither-analyzer 0.10.2

### PR DESCRIPTION
**Pull request by chenrui333**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by chenrui333** [comment]
```
  ERROR: Directory '/private/tmp/slither-analyzer--ckzg-20240413-8014-2ncrv7/c-kzg-4844-1.0.1' is not installable. Neither 'setup.py' nor 'pyproject.toml' found.
```

File: Formula/b/blst.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+class Blst < Formula
+  desc "Multilingual BLS12-381 signature library"
+  homepage "https://github.com/supranational/blst"
+  url "https://github.com/supranational/blst/archive/refs/tags/v0.3.11.tar.gz"
+  sha256 "d0a6e2a69490cc45f0a531a684a225e56fe22303665157cfa397ba5605447eb9"
+  license "Apache-2.0"
+
+  def install
+    system "./build.sh", "shared"
+    lib.install "libblst.a"
```
Comment (carlocab) [inline Formula/b/blst.rb:10]: Why are we building the shared library but not installing it?

File: Formula/b/blst.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    include.install [
+      "bindings/blst.h",
+      "bindings/blst.hpp",
+      "bindings/blst.swg",
```
Comment (carlocab) [inline Formula/b/blst.rb:14]: This doesn't look like a header file

File: Formula/b/blst.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  desc "Multilingual BLS12-381 signature library"
+  homepage "https://github.com/supranational/blst"
+  url "https://github.com/supranational/blst/archive/refs/tags/v0.3.11.tar.gz"
+  sha256 "d0a6e2a69490cc45f0a531a684a225e56fe22303665157cfa397ba5605447eb9"
+  license "Apache-2.0"
+
+  def install
+    system "./build.sh", "shared"
+    lib.install "libblst.a"
+    include.install [
```
Comment (carlocab) [inline Formula/b/blst.rb:11]: Can we install these into `include/"blst"` to reduce the risk of conflicts with other formulae?

Comment (chenrui333) [reply Formula/b/blst.rb:11]: yeah, totally.

File: Formula/b/blst.rb (hunk 2 above)
Comment (chenrui333) [reply Formula/b/blst.rb:14]: I can remove this one. 👍

File: Formula/b/blst.rb (hunk 1 above)
Comment (chenrui333) [reply Formula/b/blst.rb:10]: mostly because there is no installation step?

File: Formula/s/slither-analyzer.rb [hunk 4]
```diff
@@ -233,15 +239,11 @@ class SlitherAnalyzer < Formula
...
-  # https://github.com/crytic/slither/pull/2290
-  patch do
-    url "https://github.com/crytic/slither/commit/93cbf7d887a930e20c1ea8cf543ec2f41e84aadd.patch?full_index=1"
-    sha256 "6c6278b86c8d75fb01a50715a8891ab8a94ec6d4372092440f1cefd67fbe0940"
-  end
-
   def install
-    virtualenv_install_with_resources
+    venv = virtualenv_create(libexec, "python3.12")
+    resource("ckzg").stage { venv.pip_install "bindings/python" }
```
Comment (elopez) [inline Formula/s/slither-analyzer.rb:244]: I tried upgrading this locally; a change like this over this PR resolves the build failure

```diff
-    resource("ckzg").stage { venv.pip_install "bindings/python" }
+    resource("ckzg").stage do
+      inreplace "bindings/python/setup.py", 'include_dirs=["../../inc"', "include_dirs=[\"#{Formula["blst"].opt_include}/blst\""
+      venv.pip_install "bindings/python"
+    end
```

File: Formula/b/blst.rb (hunk 1 above)
Comment (carlocab) [reply Formula/b/blst.rb:10]: Then why build the shared library at all?

Comment (elopez) [reply Formula/b/blst.rb:10]: `shared` may be a no-op here, as the script appears to expect `-shared` with a dash: https://github.com/supranational/blst/blob/v0.3.11/build.sh#L43

**Pull request closed without merging**